flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/GenericInMemoryCatalog.java

```diff
@@ -209,11 +209,17 @@ public void alterTable(ObjectPath tablePath, CatalogBaseTable newTable, boolean
 		checkNotNull(tablePath);
 		checkNotNull(newTable);
 
-		// TODO: validate the new and old CatalogBaseTable must be of the same type. For example, this doesn't
-		//		allow alter a regular table to partitioned table, or alter a view to a table, and vice versa.
-		//		And also add unit tests.
-
 		if (tableExists(tablePath)) {
+			CatalogBaseTable oldTable = tables.get(tablePath);
+
+			// Theoretically, in-memory catalog can store any tables and views implementations
+			// thus, we need to check they are of the same class
```

Comment:
Nit: The comment doesn't seem explaining well why we want to put the check. Instead of "thus ...", we can probably say ". However, to be consistent with other catalogs, we check if...".

Reply:
there's a subtle difference between checks in in-memory catalog and HiveCatalogBase.

HiveCatalogBase can only handle HiveCatalogTable/View (HiveCatalog) or GenericCatalogTable/View (GenericHiveMetastoreCatalog), and user input is already validated in validateCatalogBaseTable(). Thus, we only need to check if the input is CatalogTable or CatalogView.

However, in-memory catalog can host any implementations of table and view, thus we need to check here that they are of the same impl class, not just whether it's a table or view.

Reply:
I guess you didn't get what I meant. I was saying the comment doesn't seem making much sense. It doesn't explain why we put checks  and why we are mentioning in-memory catalog here. There is a "thus", but I don't understand the cause/result relationship that you're trying to explain.

Reply:
I removed the comment as the code actually explains itself